src/main/java/io/druid/data/input/MapBasedRow.java

```diff
@@ -126,7 +126,8 @@ public long getLongMetric(String metric)
       return ((Number) metricValue).longValue();
     } else if (metricValue instanceof String) {
       try {
-        return Long.valueOf(((String) metricValue).replace(",", ""));
+        String s = ((String) metricValue).replace(",", "");
+        return s.matches("[-|+]?\\d+") ? Long.valueOf(s) : Double.valueOf(s).longValue();
```

Comment:
@himanshug that should work, it might also make sense to pre-compile the regexp statically to avoid the performance impact of having it compiled every time